adr/0002-reactive-rename.adoc

```diff
@@ -0,0 +1,47 @@
+= Rename Reactive extensions
+
+* Status: proposed
+* Date: 2023-10-17
+
+== Context and Problem Statement
+
+RESTEasy Reactive is the preferred REST stack in Quarkus, however in the almost 3 years since its release, we have seen
+folks of all experience misunderstand its capabilities and being led to believe that it forces reactive programming on them.
+With the advent of Virtual Threads in Java 19+, the problem becomes larger since in the minds of users, Reactive and Virtual Threads
+are two different worlds.
+
+The same problem applies to other extensions as well:
+* Reactive Messaging
+* Reactive Routes
+* REST Client reactive
+
+The common theme here is that these extensions can work equally well with reactive and blocking code, but their name confuses users.
+
+We have tried to clarify the situation with blog posts, talks and on the various communication channels, but there seems to be
+no escaping the fact that "reactive" in the name primes people in a non-desirable way.
+
+
+== Considered options
+
+We can continue with the current situation, but that feels less than ideal given the recurrence of the problem.
```

Comment:
below we talk about renaming - how about the option of introducing a new extension name which just include the existing one ?

or relocation?

Reply:
Relocations are mentioned in the proposal

Reply:
yes, so how about a new extension instead? gives no breakage, and less issues on first hunch.

Reply:
As I said below, I don't see how that is any better